packages/interactivity-router/src/assets/styles.ts

```diff
@@ -0,0 +1,85 @@
+export type StyleElement = HTMLLinkElement | HTMLStyleElement;
+
+const styleSheetCache = new Map< string, Promise< StyleElement >[] >();
+
+export const prepareStyles = (
+	doc: Document,
+	url: string = ( doc.location || window.location ).href
+): Promise< StyleElement >[] => {
+	if ( ! styleSheetCache.has( url ) ) {
+		if ( doc !== window.document ) {
+			window.document.head.appendChild(
+				window.document.createComment(
+					`@wordpress/interactivity-router: prefetched styles for ${ url }`
+				)
+			);
+		}
+		styleSheetCache.set(
+			url,
+			[ ...doc.querySelectorAll( 'style,link[rel=stylesheet]' ) ].map(
+				( element: StyleElement ) => {
+					if ( doc === window.document ) {
+						return Promise.resolve( element );
+					}
+
+					const cloned = element.cloneNode( true ) as
+						| HTMLStyleElement
+						| HTMLLinkElement;
+
+					cloned.dataset.originalMedia = cloned.media || 'all';
```

Comment:
Why do you need to store it in the DOM? Maybe it could be stored in the element instance instead.

Besides, `all` doesn't need to be stored. It can be the default value.